### PR TITLE
Add deprecation metadata to registration when it is present in catalog

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/nuget/engineering/issues/2384

### CODE REVIEW

**Comment by joelverhagen** [comment]
Can we have some unit tests that assert the actual JSON shape in the deprecated an non deprecated cases?

**Comment by loic-sharma** [comment]
image

**Comment by scottbommarito** [comment]
@joelverhagen - I found some tests (written by @dtivel) that I can add to--stay tuned!

**Comment by scottbommarito** [comment]
@joelverhagen - I finished writing the tests and also caught a bug in the process. It's ready for reviews again!

File: tests/CatalogTests/Helpers/RegistrationPackageDeprecationDetails.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        public string[] Reasons { get; }
+
+        [JsonProperty(CatalogConstants.Message)]
+        public string Message { get; }
+
+        [JsonProperty(CatalogConstants.AlternatePackage)]
+        public RegistrationPackageDeprecationAlternatePackage AlternatePackage { get; }
+    }
+
+    public class RegistrationPackageDeprecationAlternatePackage
```
Comment (joelverhagen) [inline tests/CatalogTests/Helpers/RegistrationPackageDeprecationDetails.cs]: Can we stick to one class per file please? makes github finding easier

Comment (agr) [reply tests/CatalogTests/Helpers/RegistrationPackageDeprecationDetails.cs]: Sounds like a github could use some search improvements :)

Comment (scottbommarito) [reply tests/CatalogTests/Helpers/RegistrationPackageDeprecationDetails.cs]: @anangaur - pass it on!

**Pull request merged**